### PR TITLE
FIX(make): fix incremental builds by only releasing updated headers

### PR DESCRIPTION
**Pull request by ahuber21**
## Description

This PR introduces a check using the `cmp` command in the macros responsible for publishing header files from the `include` directory to the `__release_*` directories. With this enhancement, headers are only released if their contents have changed.

This improvement addresses an issue with make's caching mechanism, which relies on file timestamps. Previously, central header files like `library_version_info.h` were released every time make was invoked, regardless of whether their contents had changed. This caused the timestamps to always be newer than those of the source files, triggering unnecessary target updates and negating the benefits of incremental builds.

With the proposed changes, only files that have actually changed (along with their dependencies) will be rebuilt, thereby preserving the efficiency of incremental builds.

Fixes https://github.com/oneapi-src/oneDAL/issues/2944
and probably also https://github.com/oneapi-src/oneDAL/issues/632

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

__does not apply - build chain only__

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

__does not apply - build chain only__

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: makefile.ver [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-#     http://www.apache.org/licenses/LICENSE-2.0
+#	 http://www.apache.org/licenses/LICENSE-2.0
```
Comment (Alexsandruss) [inline makefile.ver:8]: Tabs and spaces are mixed.

File: makefile.ver [hunk 2]
```diff
@@ -14,12 +14,12 @@
...
-UPDATE  =       0
-BUILD   =       $(shell date +'%Y%m%d')
-STATUS  =       P
-BUILDREV ?=     work
+MAJOR   =	   2025
+MINOR   =	   1
+UPDATE  =	   0
+BUILD   =	   $(shell date +'%Y%m%d')
+STATUS  =	   P
+BUILDREV ?=	 work
```
Comment (Alexsandruss) [inline makefile.ver:22]: Same here.

**Comment by ahuber21** [comment]
Thanks @Alexsandruss. It was still a draft, but the latest changes seem to fix the issue and don't add any new spaces.
@sterrettm2 could you check out the latest commit `3cbe3ac` and let me know if incremental build still works for you?

File: makefile.ver [hunk 3]
```diff
@@ -33,8 +33,8 @@ VERSION_DATA_FILE = $(WORKDIR)/daal_version_gen.h
 
 # content of version data file to be included in product
 define DAAL_VERSION_DATA
-// DO NOT EDIT: file is auto-generated on build time
-// DO NOT PUT THIS FILE TO SVN: file is auto-generated on build time
+// DO NOT PUT THIS FILE TO VERSIONING: file is auto-generated on build time
+// Product version is specified in `makefile.ver` file
 // Product version is specified in src/makefile.ver file
```
Comment (Alexsandruss) [inline makefile.ver:38]: Don't these two lines say the same?

File: makefile.ver [hunk 4]
```diff
@@ -61,9 +61,17 @@ $(VERSION_DATA_FILE): | $(WORKDIR)/.
...
 		-e "s/\($${mark}_UPDATE__\).*/\1 $(UPDATE)$(sed.eol)/" \
-		-e "s/\($${mark}_STATUS__\).*/\1 \"$(STATUS)\"$(sed.eol)/" $$file && \
-        rm -f $$file.bak
+		-e "s/\($${mark}_STATUS__\).*/\1 \"$(STATUS)\"$(sed.eol)/" \
+		$$file > $$file.tmp; \
+	if ! cmp -s $$file.tmp $$file; then \
+		mv $$file.tmp $$file; \
+	else \
+		rm $$file.tmp; \
+	fi
```
Comment (Alexsandruss) [inline makefile.ver:77]: Line break at end of file is missing
```suggestion
	fi

```

File: makefile.ver (hunk 3 above)
Comment (ahuber21) [reply makefile.ver:38]: Thanks for noticing, was a problem with rebasing.

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**